### ISSUE TITLE
Error when running cost estimation on environment with azurerm_mssql_database with standard DTU SKU

### ISSUE THREAD

**Issue by holzro** [infracost/infracost#755]
I get the below error when trying to run a cost estimate on an environment with an azurerm_mssql_database with a SKU in the standard DTU tier (i.e. S0, S1, S2)

```
❯ infracost breakdown --path ./
Detected Terraform directory at ./
  ✔ Running terraform plan

Error: An unexpected error occurred

runtime error: index out of range [0] with length 0
goroutine 1 [running]:
runtime/debug.Stack(0x0, 0xc0000a45b0, 0xc000783a80)
	runtime/debug/stack.go:24 +0x9f
main.handleUnexpectedErr(0xc0001a7930, 0x1752800, 0xc0003cddd0)
	github.com/infracost/infracost/cmd/infracost/main.go:220 +0x45
main.main.func1(0xc000159ec8, 0xc0001a7930, 0xc000132240)
	github.com/infracost/infracost/cmd/infracost/main.go:39 +0xa5
panic(0x1752800, 0xc0003cddd0)
	runtime/panic.go:965 +0x1b9
github.com/infracost/infracost/internal/providers/terraform/azure.NewAzureMSSQLDatabase(0xc000292e80, 0x0, 0xc000a1de7f)
	github.com/infracost/infracost/internal/providers/terraform/azure/mssql_database.go:28 +0x2dee
github.com/infracost/infracost/internal/providers/terraform.(*Parser).createResource(0xc000159710, 0xc000292e80, 0x0, 0x0)
	github.com/infracost/infracost/internal/providers/terraform/parser.go:66 +0x174
github.com/infracost/infracost/internal/providers/terraform.(*Parser).parseJSONResources(0xc000159710, 0x3f100, 0x1c4cb48, 0x0, 0x0, 0xc00007c0c0, 0x5, 0xc000a14000, 0x3f153, 0x0, ...)
	github.com/infracost/infracost/internal/providers/terraform/parser.go:114 +0x4da
github.com/infracost/infracost/internal/providers/terraform.(*Parser).parseJSON(0xc000159710, 0xc0009c8000, 0x3f153, 0x4c000, 0xc00007c0c0, 0x0, 0x1, 0x1, 0x7, 0x0, ...)
	github.com/infracost/infracost/internal/providers/terraform/parser.go:135 +0x892
github.com/infracost/infracost/internal/providers/terraform.(*DirProvider).LoadResources(0xc000094240, 0xc00007c0c0, 0x0, 0xc00007c0c0, 0x0)
	github.com/infracost/infracost/internal/providers/terraform/dir_provider.go:104 +0x145
main.runMain(0xc0000c4780, 0xc0001a7930, 0x0, 0x0)
	github.com/infracost/infracost/cmd/infracost/run.go:78 +0x3be
main.breakdownCmd.func1(0xc0000c4780, 0xc0000a6180, 0x0, 0x2, 0x0, 0x0)
	github.com/infracost/infracost/cmd/infracost/breakdown.go:40 +0x106
github.com/spf13/cobra.(*Command).execute(0xc0000c4780, 0xc0000a6160, 0x2, 0x2, 0xc0000c4780, 0xc0000a6160)
	github.com/spf13/cobra@v1.1.3/command.go:852 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xc0000c4000, 0x351, 0xc000159ee8, 0x7)
	github.com/spf13/cobra@v1.1.3/command.go:960 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.1.3/command.go:897
main.main()
	github.com/infracost/infracost/cmd/infracost/main.go:171 +0xc19

Environment:
Infracost v0.8.7
```

**Comment by alikhajeh1** [comment]
This seems to be a "we can't find the location" issue too, `azurerm_mssql_database` requires `azurerm_sql_server` and we use the server's `location`. Is the `azurerm_sql_server` defined in another module?

**Comment by holzro** [comment]
MSSQL server is defined in the same environment, this specific environment is not using any modules

**Comment by alikhajeh1** [comment]
Interesting! Can you possibly paste the `azurerm_mssql_database` and the `azurerm_sql_server` and any resource group they refer to? I'm working on this just now

**Comment by holzro** [comment]
Here you go 

```
resource "azurerm_resource_group" "rg" {
  name     = var.rg_name
  location = var.pri_location
  tags     = local.common_tags
}

data "azurerm_mssql_server" "server" {
  name                = var.server_name
  resource_group_name = var.server_rg
}

resource "azurerm_mssql_database" "database" {
  name      = var.database_name
  server_id = data.azurerm_mssql_server.server.id
  sku_name  = "S1"
  tags      = local.common_tags
}
```

**Comment by alikhajeh1** [comment]
@holzro we pushed a fixed for this in v0.9.0 (released just now), feel free to comment if there are still issues (plz run with `--log-level=warn` if there are issues and include any interesting warnings).
